src/python/bot/fuzzers/syzkaller/engine.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fuzzing engine interface."""
+
+from bot.fuzzers import engine
+from bot.fuzzers import engine_common
+from bot.fuzzers import libfuzzer
+from bot.fuzzers import utils as fuzzer_utils
+from bot.fuzzers.syzkaller import fuzzer
+from builtins import object
+from metrics import profiler
+from system import environment
+import os
+
+ENGINE_ERROR_MESSAGE = 'syzkaller: engine encountered an error'
+_ENGINES = {}
+
+
+class SyzkallerError(Exception):
+  """Base exception class."""
+
+
+class SyzkallerOptions(engine.FuzzOptions):
+  """Represents options passed to the engine. Can be overridden to provide more
+  options."""
+
+  def __init__(self, corpus_dir, arguments, strategies, fuzz_corpus_dirs,
+               extra_env):
+    super(SyzkallerOptions, self).__init__(corpus_dir, arguments, strategies)
+    self.fuzz_corpus_dirs = fuzz_corpus_dirs
+    self.extra_env = extra_env
+
+
+class SyzkallerEngine(object):
+  """Syzkaller fuzzing engine implementation."""
+
+  @property
+  def name(self):
+    return 'syzkaller'
+
+  def get_name(self):
+    return 'syzkaller'
+
+  def prepare(self, corpus_dir, target_path, build_dir):
+    """Prepare for a fuzzing session, by generating options. Returns a
+    FuzzOptions object.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+
+    Returns:
+      A FuzzOptions object.
+    """
+    del build_dir
+    arguments = fuzzer.get_arguments(target_path)
+
+    # Add strategies here
```

Comment:
nit: convert to a TODO? Or if this doesn't make sense in the context of syzkaller, just remove the comment?

Reply:
Converted to TODO